Source/WebCore/rendering/LegacyInlineBox.cpp

```diff
@@ -190,6 +192,23 @@ void LegacyInlineBox::adjustPosition(float dx, float dy)
         downcast<RenderBox>(renderer()).move(LayoutUnit(dx), LayoutUnit(dy));
 }
 
+void LegacyInlineBox::adjustBlockDirectionPosition(float delta, bool adjustStaticPosition)
+{
+    if (isHorizontal())
+        LegacyInlineBox::adjustPosition(0, delta);
+    else
+        LegacyInlineBox::adjustPosition(delta, 0);
+
+    if (adjustStaticPosition && renderer().isOutOfFlowPositioned()) {
+        RenderBox& box = static_cast<RenderBox&>(renderer());
```

Comment:
```suggestion
        auto& box = static_cast<RenderBox&>(renderer());
```
but also, why are we using static_cast here?
Also, this may need a smart pointer

Reply:
Please use a CheckedRef.